service/storage_proxy.hh

```diff
@@ -482,8 +482,10 @@ public:
         }
         return next;
     }
-    void init_messaging_service(netw::messaging_service&, gms::gossiper&, migration_manager&);
-    future<> uninit_messaging_service();
+
+    // Start/stop the remote part of `storage_proxy` that is required for performing distributed queries.
+    void start_remote(netw::messaging_service&, gms::gossiper&, migration_manager&);
+    future<> stop_remote();
```

Comment:
The `storage_proxy` initialization doesn't become simpler with that, preferably it should only have `start`, `stop` and `shutdown` (or `drain`) methods, but anyway it's better than it was before

Reply:
`storage_proxy` init will have to be 2-phase with this approach, one phase for local queries, another for remote queries.

Although the one for local queries maybe doesn't need anything else than the constructor?